omod/src/main/resources/config.xml

```diff
@@ -86,37 +90,38 @@
 		<property>${project.parent.artifactId}.openempi.globalIdentifierDomainId</property>
 		<defaultValue></defaultValue>
 		<description>
-			Specifies main identifier on MPI server (e.g. "xyz" for OpenEMPI)
+			Specifies main identifier on MPI server (e.g. "xyz" for OpenEMPI) (GSOC OpenEMPI Specific Implementation)
 		</description>
 	</globalProperty>
 
 	<globalProperty>
 		<property>${project.parent.artifactId}.mpi.implementation</property>
-		<defaultValue></defaultValue>
+		<defaultValue>registrationcore.mpi.implementation.PixPdq</defaultValue>
 		<description>
-			Which MPI to should we connect to? Specify a Spring bean name, or leave blank to disable MPI integration
+			Which MPI to should we connect to? Specify a Spring bean name, or leave blank to disable MPI integration. Options include:
+			"registrationcore.mpi.implementation.PixPdq", "registrationcore.mpi.implementation.OpenEMPI".
```

Comment:
This GP is confusing, PixPdq is the profile and OpenMPI implements this profile, I'd expect the MPI integration to be generic and work with any software that implements the profile.

Reply:
This is true, it is confusing, but understanding the history of how this functionality was developed can shed some light on what is happening. The spring bean "registrationcore.mpi.implementation.OpenEMPI" was built by @approce during GSOC 2015 and is a rest based OpenEMPI specific implementation. Following that the SolDevelo team built a more generic PIX/PDQ profile that implemented the same interface, so hypothetically this PIXPDQ profile can work with any EMPI that accepts PIXPDQ as you mentioned.

Reply:
Then I think we need to get rid of the openempi specific implementation because it's redundant and then get rid of the GP too.

Reply:
Mmmmm so you think we should completely cut out this entire implementation? https://github.com/shaoyuancc/openmrs-module-registrationcore/tree/master/api/src/main/java/org/openmrs/module/registrationcore/api/mpi/openempi
at this early stage in development of this MPI functionality I don't necessarily feel like we should be cutting this out because we are still facing issues with the various MPIs we are trying to connect to and don't have full configurable functionality, and if in the future more blocks are found it might be possible that we would go back and explore the REST implementation? 

in the future if other profiles are built, for example fhir or hl7v3 or a generic REST one then this would be the GP to redirect to it. so I do think that we should keep it...

Reply:
But in that case the GP would no longer be specifying the MPI implementation but the data exchange format so we can never reuse the same GP

Reply:
Ok, had a discussion with @mseaton, seeing that this implementation was already in the code before this, there might be people that are dependent on that so I think if we want to remove this direct rest OpenEMPI specific implementation we should create that ticket and have that discussion separately after this is merged :) Does that sound like a good plan?

Reply:
That's fine, but I still think that if some people are dependent on this, it's those that are running the module from where you are copying this code, so my assumption is that if they still need this they continue using that other module instead, when they are ready to make the switch they stop depending on the GP.

With that said, if we are implementing a standard profile, why and how comes we have code specific to an specific implementation of this standard? To me it raises a red flag that there is something wrong we are doing.

Reply:
No, this openEMPI specific implementation wasn't created in the isanteplus fork. It was created by @approce in __THIS BRANCH__ for the 2015 GSOC project! So it has been in registration core for a while. 
"implementing a standard profile, why and how comes we have code specific to an specific implementation of this standard?" -  So as previously explained, we have one implementation of a standard profile (PixPdq) and we have one implementation specific to OpenEMPI (GSOC project built by roman and that has been in registrationcore for a couple of years) these are two separate things! So if you feel like we should remove the OpenEMPI specific rest implementation then we can have that discussion separately.

Reply:
PIXPDQ is the profile and OpenEMPI implements this profile, so there should be no such thing as a PIXPDQ and OpenMPI implementation because one is already the other anyways, does this make sense?

If this was done in a branch as part of GSoC, is there any release that was made off this branch? If not, which I guess is the case, then nobody can be dependent on it in the first place.

Reply:
If the module is authored to implement the PIXPDQ profile, it should automatically work with OpenEMPI, if it not it a red flag that the code is not doing the correct thing and we need to address that rather than writing an OpenMPI implementation.

Reply:
"is there any release that was made off this branch?" - All the commits were done to master, e.g. https://github.com/openmrs/openmrs-module-registrationcore/commit/85aa6cf5bda07b1751918847a441002f0fc03d0e if you take a look at the commit history you will see all this OpenEMPI specific code is from 2015, it was just refactored by the soldevelo team so that they could implement their PixPdq implementation. 

it __does__ automatically work with OpenEMPI...